sgxs-tools/src/sgx_detect/imp/windows.rs

```diff
@@ -0,0 +1,46 @@
+use std::collections::HashMap;
+use std::ffi::OsString;
+use std::fs::{File, read_dir};
+use std::io::{BufRead, BufReader, ErrorKind, Read, Seek, SeekFrom};
+use std::os::windows::ffi::OsStringExt;
+use std::path::PathBuf;
+use std::process::Command;
+
+use byteorder::{ReadBytesExt, LE};
+use failure::{Error, Fail, ResultExt};
+
+use crate::DetectError;
+use crate::interpret::{AesmStatus, KmodStatus};
+use winapi::um::winbase::GetFirmwareEnvironmentVariableA;
+use winapi::shared::minwindef::DWORD;
+use winapi::um::winnt::{LPCSTR, PVOID};
+
+pub fn rdmsr(address: u64) -> Result<u64, Error> {
+    bail!("Currently not implemented")
+}
+
+pub fn read_efi_var(name: &str, guid: &str) -> Result<Vec<u8>, Error> {
+    let mut env: Vec<u8> = vec![0; 1024];
+    let ret: DWORD = unsafe {
+        GetFirmwareEnvironmentVariableA(
+            name.as_ptr() as _,
+            guid.as_ptr() as _,
```

Comment:
The types of these parameters are `LPCSTR`. So you need to convert the Rust strings to C strings before passing them. Use `std::ffi::CString`.